### DISCUSSION TITLE
Populating Flashcard Content from Notes

### DISCUSSION THREAD

**Discussion by suprafly** [st3v3nmw/obsidian-spaced-repetition#725] [General]
Hey I just wanted to say thanks for the amazing work ya'll have been doing on this plugin. I just got started with it over the weekend and I am really excited about being able to completely replace Anki (finally!).

I am using this for language learning and one feature I realized I would love would be to be able to populate flashcards from notes. For example, suppose I have the phase, "Selamat Pagi", Indonesian for "Good morning!". And suppose I have also learned the same phrase in Swedish, "God morgon!".

I would love to be able to turn my notes for those phrases into a flashcard. Spitballing, conceptually I think it could work something like this,

```
Indonesian Vocab Note
---
tags: flashcards
---

[[Selamat pagi]] <> [[God morgon]]
```

That would create bi-directional flashcards and the content of the flashcards could be whatever is in those notes that are backlinked.

This gives a few of interesting benefits,
1. It allows for flashcards to be created out of content that already exists
2. It allows flashcard content to become first class content in Obsidian which is then referenced.
3. It allows for flashcards to link knowledge in a way that Anki never could.

A bit more on point 3 - with language learning, (and I believe this applies to non-language learning as well) flashcards are like a snapshot of the neural connections between words. So the connection between "Selamat pagi" <> "God morgon" <> "Good morning" is a window into different ways of seeing the same concept. The concept of being "well early in the day" is being overloaded, and being able to link discrete notes together into flashcards is a step towards pulling this abstraction out and into reality.

Suppose I come along and decide to learn Spanish. I add a note for "Buen día" and link it, "Buen día" <> "Selamat pagi". Now I am making a connection between my Indonesian concept and Spanish concept, so it is reinforcing the interconnectedness, and all of this should become visible from Obsidian's graph view.

One thing that might be worth considering is that you may want only a subset of information on those backlinked notes to appear in a flashcard. There is probably a clever way of structuring the notes to specify this via the YAML front matter.

I am a software developer, and if this idea seems fruitful I would be more than happy to do a spike on it and see what it might look like.

[I did see a similar discussion opened, but when I read it the author seemed to be approaching a similar concept from a different direction.](https://github.com/st3v3nmw/obsidian-spaced-repetition/discussions/724) I wanted to start this one because I believe that they are actually quite different.

Thanks again for all of your hard work!

**Comment by ronzulu** [comment]
Hi @suprafly 

This can (kind of) be achieved using Obsidian transclusion, for example:

Raw text
```
 #flashcards/test/725

![[Info 1]]:::![[Info 2]]
```

This shows as the following card during review:
![image](https://github.com/user-attachments/assets/eb9f08d6-f662-4faf-a7e7-3ac81fb0da98)

One characteristic of this approach is that in Obsidian it is rendered in the expanded form:
![image](https://github.com/user-attachments/assets/7f9ec205-62f6-401f-a00c-5ff26c98b8c6)

Which some users may prefer, and others not.

Does this suggestion suit you?

Regards
Ronny
